fee-provider/src/main/scala/org/bitcoins/feeprovider/HttpFeeRateApi.scala

```diff
@@ -0,0 +1,74 @@
+package org.bitcoins.feeprovider
+
+import java.time.{Duration, Instant}
+
+import akka.actor.ActorSystem
+import akka.http.scaladsl.Http
+import akka.http.scaladsl.model.{HttpRequest, Uri}
+import akka.stream.ActorMaterializer
+import akka.util.ByteString
+import org.bitcoins.core.api.FeeRateApi
+import org.bitcoins.core.util.TimeUtil
+import org.bitcoins.core.wallet.fee.FeeUnit
+
+import scala.concurrent.{ExecutionContextExecutor, Future}
+import scala.util.Try
+
+object HttpFeeRateApi {
+
+  def makeApiCall(uri: Uri)(implicit system: ActorSystem): Future[String] = {
+    implicit val mat: ActorMaterializer = ActorMaterializer.create(system)
+    implicit val ec: ExecutionContextExecutor = mat.executionContext
+    Http()
+      .singleRequest(HttpRequest(uri = uri))
+      .flatMap(
+        response =>
+          response.entity.dataBytes
+            .runFold(ByteString.empty)(_ ++ _)
+            .map(payload => payload.decodeString(ByteString.UTF_8))
+      )
+  }
+}
+
+abstract class HttpFeeRateApi extends FeeRateApi {
+  implicit val system: ActorSystem
+
+  def uri: Uri
+
+  def converter: String => Try[FeeUnit]
+
+  def getFeeRate: Future[FeeUnit] = {
+    HttpFeeRateApi
+      .makeApiCall(uri)
+      .flatMap(ret => Future.fromTry(converter(ret)))(system.dispatcher)
+  }
+}
+
+abstract class CachedHttpFeeRateApi extends HttpFeeRateApi {
+
+  private var cachedFeeRateOpt: Option[(FeeUnit, Instant)] = None
+
+  val cacheDuration: Duration = Duration.ofMinutes(5)
```

Comment:
Probably should be private, and also configurable in a 
```
bitcoin-s { 
  fee-provider { 
    cache-duration = 5.minutes 
  }
}
```

Reply:
I think having it public is better, it allows other people using the library to set their own `cacheDuration`